src/main/java/com/github/simbo1905/nfp/srs/FileRecordStore.java

```diff
@@ -1543,8 +1555,30 @@ private RecordHeader findFreeRecord(int payloadLength) throws IOException {
         logger.log(
             Level.FINEST,
             () -> String.format("DEBUG findFreeRecord: created newRecord=%s", finalNewRecord2));
-        updateFreeSpaceIndex(next);
-        writeRecordHeaderToIndex(next);
+        
+        // Fix: Reduce the capacity of the original record to complete the split
```

Comment:
Do not write `// Fix: ` in a comment. Do not have double single-line comments on local local variable. speak less to say more.